### DISCUSSION TITLE
Filtering with inner join returns unexpected results

### DISCUSSION THREAD

**Discussion by peperoli** [supabase/supabase#17799] [Questions] [answered]
```js
const filterMembers = null

const query = supabase
  .from('meetings')
  .select('*, members:join_members!inner(id)')

if (filterMembers) {
  query = query.in('members.id', filterMembers)
}

const { data, error } = await query
```

This example similar to the one from the docs works as expected as long as every meeting has at least one member.
But if there are no members for a meeting in the join_members table, then this meeting is **filtered out entirely**.

I'm not really sure if this is a bug or expected behaviour, but if it is expected I would appreciate help in getting all the meetings regardless if they have a row in the join table or not.

**Comment by GaryAustin1** [comment, marked as answer]
When you add !inner you are saying you want the parent table results filtered also.  Leave that part out and you will get all meetings and if there are members matching your list they will show up in the members list.

**Reply by peperoli** [reply]
Yeah it makes sense. It's just that initially I don't want to filter the meetings at all because I want to see all of them. That's why I set the `in()` filter conditionally. I don't think there's a way to use `!inner` conditionally.

**Reply by GaryAustin1** [reply]
The select is just a string.  Set the string as you want for each case based on if you are going to add the filter.

**Reply by peperoli** [reply]
I feel stupid now. Thanks for your answers :)
